### PR TITLE
CloudWatch: Add missing GameLift metrics

### PR DESCRIPTION
**Pull request by Ciara-Cloud**
**What is this feature?**

I have added the following metrics for standard GameLift Fleet IQ metrics which are currently not available in Grafana 
AvailableGameServers
DrainingUtilizedGameServers
DrainingAvailableGameServers
UtilizedGameServers

**Why do we need this feature?**
This is to follow up on the issue I opened here https://community.grafana.com/t/aws-grafana-gamelift-hard-coded-metrics/100341

**Who is this feature for?**

Amazon Managed Grafana dashboard for GameLift Fleet IQ metrics

### CODE REVIEW

File: pkg/tsdb/cloudwatch/constants/metrics.go [hunk 1]
```diff
@@ -337,7 +337,7 @@ var NamespaceMetricsMap = map[string][]string{
...
+	"AWS/GameLift":             {"ActivatingGameSessions", "ActiveGameSessions", "ActiveInstances", "ActiveServerProcesses", "AvailableGameServers","AvailableGameSessions", "AverageWaitTime", "CurrentPlayerSessions", "CurrentTickets", "DesiredInstances","DrainingAvailableGameServers","DrainingUtilizedGameServers", "FirstChoiceNotViable", "FirstChoiceOutOfCapacity", "GameSessionInterruptions", "HealthyServerProcesses", "IdleInstances", "InstanceInterruptions", "LowestLatencyPlacement", "LowestPricePlacement", "MatchAcceptancesTimedOut", "MatchesAccepted", "MatchesCreated", "MatchesPlaced", "MatchesRejected", "MaxInstances", "MinInstances", "PercentAvailableGameSessions", "PercentHealthyServerProcesses", "PercentIdleInstances", "Placement", "PlacementsCanceled", "PlacementsFailed", "PlacementsStarted", "PlacementsSucceeded", "PlacementsTimedOut", "PlayerSessionActivations", "PlayersStarted", "QueueDepth", "RuleEvaluationsFailed", "RuleEvaluationsPassed", "ServerProcessAbnormalTerminations", "ServerProcessActivations", "ServerProcessTerminations", "TicketsFailed", "TicketsStarted", "TicketsTimedOut", "TimeToMatch", "TimeToTicketSuccess","UtilizedGameServers"},
```
Comment (jangaraj) [inline pkg/tsdb/cloudwatch/constants/metrics.go:340]: I would say a few spaces is missing. IMHO it should be:
`"AWS/GameLift":             {"ActivatingGameSessions", "ActiveGameSessions", "ActiveInstances", "ActiveServerProcesses", "AvailableGameServers", "AvailableGameSessions", "AverageWaitTime", "CurrentPlayerSessions", "CurrentTickets", "DesiredInstances", "DrainingAvailableGameServers", "DrainingUtilizedGameServers", "FirstChoiceNotViable", "FirstChoiceOutOfCapacity", "GameSessionInterruptions", "HealthyServerProcesses", "IdleInstances", "InstanceInterruptions", "LowestLatencyPlacement", "LowestPricePlacement", "MatchAcceptancesTimedOut", "MatchesAccepted", "MatchesCreated", "MatchesPlaced", "MatchesRejected", "MaxInstances", "MinInstances", "PercentAvailableGameSessions", "PercentHealthyServerProcesses", "PercentIdleInstances", "Placement", "PlacementsCanceled", "PlacementsFailed", "PlacementsStarted", "PlacementsSucceeded", "PlacementsTimedOut", "PlayerSessionActivations", "PlayersStarted", "QueueDepth", "RuleEvaluationsFailed", "RuleEvaluationsPassed", "ServerProcessAbnormalTerminations", "ServerProcessActivations", "ServerProcessTerminations", "TicketsFailed", "TicketsStarted", "TicketsTimedOut", "TimeToMatch", "TimeToTicketSuccess", "UtilizedGameServers"},`

**Comment by jangaraj** [comment]
I would say also dimensions for `AWS/GameLift`, should be updated. IMHO they should be:
```
"AWS/GameLift":                {"Location", "FleetId", "MetricGroup", "QueueName", "ConfigurationName", "ConfigurationName-RuleName", "InstanceType", "OperatingSystem", "GameServerGroup"},
```

**Comment by iwysiu** [comment]
Hi @doitwithciara , thanks for your contribution!
It looks good, but I was going through the amazon docs, and I noticed these metrics that are also missing:
RecycledInstances
ConcurrentActivatableGameSessions
TimeToTicketCancel
PercentUtilizedGameServers
GameServerInterruptions

**Comment by jangaraj** [comment]
Dimensions are still not correct. They have been wrong in the AWS doc in the past, so they are wrong also here atm. But AWS fixed the dec (https://repost.aws/questions/QUUvVFf1MKRF-kmgPtsoDk1Q/incorrect-documentation-for-dimensions-for-gamelift-metrics-in-cloudwatch)

**Review by fridgepoet — APPROVED** [review]
Thanks @doitwithciara I was able to see the GameLift Fleet IQ metrics on the AWS doc you linked.
Will let @iwysiu give the final approval.

(You probably saw this already: Tthe go linter might be complaining about an extra space after "Location" in the Dimensions which I'm seeing after running `gofmt -w metrics.go` in that directory. Not sure if there is anything else to fix.)

**Comment by fridgepoet** [comment]
Hi @doitwithciara do you still want to work on this one? By the way, you'll have to sign https://cla-assistant.io/grafana/grafana?pullRequest=72805

**Comment by fridgepoet** [comment]
Hi @Ciara-Cloud do you still want to work on this one? By the way, you'll have to sign https://cla-assistant.io/grafana/grafana?pullRequest=72805

**Comment by fridgepoet** [comment]
Replaced with https://github.com/grafana/grafana/pull/76102 due to inactivity

**Pull request closed without merging**